syntax = "proto3";


message procResult {
    string result = 1;
}

message idRequest {
    string id = 1;
}

message loginRequest {
    string id = 1;
    string pw = 2;
}

message chngPwRequest {
    string id = 1;
    string pw1 = 2;
    string pw2 = 3;
}

message infoData {
    string id = 1;
    string gender = 2;
    int32 age = 3;
    string job = 4;
    string relation = 5;
}

service infoManager {
    rpc ChkInfo (idRequest) returns (procResult);
    rpc SetInfo (infoData) returns (procResult);
    rpc GetInfo (idRequest) returns (infoData);
    rpc SignUp (loginRequest) returns (procResult);
    rpc LogIn (loginRequest) returns (procResult);
    rpc ChngPw (chngPwRequest) returns (procResult);
    rpc DelAccount (loginRequest) returns (procResult);
}

message cardRequest {
    int32 n1 = 1;
    int32 n2 = 2;
    int32 n3 = 3;
}

message card {
    int32 n = 1;
    string eng = 2;
    string kor = 3;
    string meaning = 4;
    int32 reverse = 5;
}

message cardResponse {
    card c1 = 1;
    card c2 = 2;
    card c3 = 3;
}

message saveRequest {
    string id = 1;
    string content = 2;
}

message record {
    string date = 1;
    string content = 2;
}

message tarotRecords {
    repeated record records = 1;
}

service TarotPlayer {
    rpc GetCards (cardRequest) returns (cardResponse);
    rpc SaveTarot (saveRequest) returns (procResult);
    rpc LoadTarot (idRequest) returns (tarotRecords);
}